devito/types/basic.py

```diff
@@ -490,6 +491,11 @@ def _cache_key(cls, *args, **kwargs):
         # From the kwargs
         key.update(kwargs)
 
+        # Any missing __rkwargs__ along with their default values
+        params = inspect.signature(cls.__init_finalize__).parameters
```

Comment:
Ouch how can we end up in such a weird spot

Reply:
try the newly added tests without this patch 😬 I don't remember the details, but basically caching bypassed because a different cache key gets computed

Reply:
Sure but I don't get why this needs this elaborate inspect instead of just having StencilDimension implement `_cache_key` and add step/spacing

Reply:
Im not sure what makes you think it's due to StencilDimension? the test maybe ? but it was not just that. Maybe it emerged from there, but the problem is way more general. In fact, IIRC the issue was the presence/absence of the `is_const` flag, which pops up after reconstruction but it's not part of the key (without this patch) at first instantiation